libr/core/cmd_cmp.c

```diff
@@ -58,103 +59,203 @@ R_API RCoreCmpWatcher *r_core_cmpwatch_get(RCore *core, ut64 addr) {
 	return NULL;
 }
 
-R_API int r_core_cmpwatch_add(RCore *core, ut64 addr, int size, const char *cmd) {
+R_API bool r_core_cmpwatch_add(RCore *core, ut64 addr, int size, const char *cmd) {
 	RCoreCmpWatcher *cmpw;
-	if (size < 1) {
-		return false;
-	}
+	bool found = false;
+	r_return_val_if_fail (core && cmd && size > 0, false);
+
 	cmpw = r_core_cmpwatch_get (core, addr);
 	if (!cmpw) {
 		cmpw = R_NEW (RCoreCmpWatcher);
 		if (!cmpw) {
 			return false;
 		}
 		cmpw->addr = addr;
+	} else {
+		free (cmpw->odata);
+		free (cmpw->ndata);
+		free (cmpw->cmd);
```

Comment:
```suggestion
		free (cmpw->cmd);
		free (cmpw);
		return false;
```

Reply:
we're overwriting the watcher, not removing it